File: bot/src/main/java/net/slimediamond/atom/command/discord/DiscordCommandSender.java
package net.slimediamond.atom.command.discord;

import net.dv8tion.jda.api.entities.User;
import net.slimediamond.atom.Atom;
import net.slimediamond.atom.command.CommandSender;
import net.slimediamond.atom.data.Database;

import java.sql.SQLException;

// TODO: Extend JDA command sender
public class DiscordCommandSender implements CommandSender {
    private String name;
    private User user;
    private Database database;

    public DiscordCommandSender(String name, User user) {
        this.name = name;
        this.user = user;

        this.database = Atom.getServiceManager().getInstance(Database.class);
    }

    @Override
    public String getName() {
        return this.name;
    }

    @Override
    public boolean isAdmin() throws SQLException {
        return database.isDiscordAdminByID(this.getId());
    }

    public Long getId() {
        return this.user.getIdLong();
    }

    public User getRaw() {
        return this.user;
    }
}

File: bot/src/main/kotlin/net/slimediamond/atom/api/command/platforms/discord/DiscordCommandSender.kt
package net.slimediamond.atom.api.command.platforms.discord

import net.slimediamond.atom.api.command.CommandSender
import net.slimediamond.atom.api.discord.entities.User
import net.slimediamond.atom.api.messaging.RichText
import net.slimediamond.atom.storage.dao.UserDao

class DiscordCommandSender(private val user: User) : CommandSender {

    override val name: String
        get() = user.displayName
    override val userDao: UserDao?
        get() = UserDao.getFromDiscord(user).orElse(null)

    override fun hasPermission(permission: String): Boolean {
        return UserDao.getFromDiscord(user)
            .map { user -> user.hasPermission(permission) }
            .orElse(false)
    }

    override fun sendMessage(message: String) {
        user.sendMessage(message)
    }

    override fun sendMessage(message: RichText) {
        user.sendMessage(message)
    }

}